File: transport_marchandises_web/src/models/Utilisateur.js
export class Utilisateur {
  constructor({
    id = null,
    nom = "",
    prenom = "",
    email = "",
    password = "",
    telephone = "",
    adresse = "",
    role = "",
    type_utilisateur = "",
    image_profile = null,
    urlImageProfile = "",
  } = {}) {
    this.id = id;
    this.nom = nom;
    this.prenom = prenom;
    this.email = email;
    this.password = password;
    this.telephone = telephone;
    this.adresse = adresse;
    this.role = role;
    this.type_utilisateur = type_utilisateur;
    this.image_profile = image_profile;
    this.urlImageProfile = urlImageProfile;
  }

  /** Convertit l'objet en JSON prêt à envoyer à l'API */
  toJSON() {
    return {
      id: this.id,
      nom: this.nom,
      prenom: this.prenom,
      email: this.email,
      password: this.password,
      telephone: this.telephone,
      adresse: this.adresse,
      role: this.role,
      type_utilisateur: this.type_utilisateur,
      image_profile: this.image_profile,
    };
  }

  /** Crée une instance Utilisateur à partir d'une réponse JSON */
  static fromJSON(data) {
    if (!data) return null;
    return new Utilisateur({
      id: data.id,
      nom: data.nom,
      prenom: data.prenom,
      email: data.email,
      telephone: data.telephone,
      adresse: data.adresse,
      role: data.role,
      type_utilisateur: data.type_utilisateur,
      urlImageProfile: data.image_profile,
    });
  }
}
export default Utilisateur;

File: transport_marchandises/lib/models/utilisateur.dart
import 'package:image_picker/image_picker.dart';
import 'dart:typed_data';

class Utilisateur {
  int? id;
  String? nom;
  String? prenom;
  String? email;
  String? password;
  String? telephone;
  String? adresse;
  String? role;
  String? type_utilisateur;
  Uint8List? image_profile;
  String? urlImageProfile;

  Utilisateur({
    this.id,
    this.nom,
    this.prenom,
    this.email,
    this.password,
    this.telephone,
    this.adresse,
    this.role,
    this.type_utilisateur,
    this.image_profile,
    this.urlImageProfile,
  });

  Map<String, dynamic> toJSON() {
    return {
      'id': id,
      'nom': nom,
      'prenom': prenom,
      'email': email,
      'password': password,
      'telephone': telephone,
      'adresse': adresse,
      'role': role,
      'type_utilisateur': type_utilisateur,
      'image_profile': image_profile,
    };
  }

  static Utilisateur? jsonToUser(data) {
    if (data == null) return null;

    return Utilisateur(
      id: data["id"],
      nom: data['nom'] as String?,
      prenom: data['prenom'] as String?,
      email: data['email'] as String?,
      telephone: data['telephone'] as String?,
      adresse: data['adresse'] as String?,
      role: data['role'] as String?,
      type_utilisateur: data['type_utilisateur'] as String?,
      urlImageProfile: data['image_profile'] as String?,
    );
  }
}
